tests/functional/test_show.py

```diff
@@ -79,14 +79,43 @@ def test_find_package_not_found():
     assert len(list(result)) == 0
 
 
+def test_report_single_not_found(script):
+    """
+    Test that a single unmatched package is reported not found.
+    """
+    # The following should report an error as there no result to print hence
+    # expect_error=True which implies allow_stderr_warning=True
+    result = script.pip('show', 'abcd-3', expect_error=True)
+    assert 'WARNING: Package(s) not found:' in result.stderr
+    # Note that this should report the un-canonicalized name
+    assert 'abcd-3' in result.stderr
+
+
+def test_report_mixed_not_found(script):
+    """
+    Test that a unmatched package is reported not found but matches are found.
+    """
+    # Note that the following should return pass as 1 or more packages
+    # are found but should have warning in stderr for not found so needs
+    # allow_stderr_warning=True to aviod a failure.
+    result = script.pip('show', 'abcd3', 'A-B-C', 'pip',
```

Comment:
minor style nitpick: where we aren't matching existing code, I think the [black](https://black.readthedocs.io/en/stable/the_black_code_style.html) code style is preferred, which would look like:

```python
result = script.pip(
    'show', 'abcd3', 'A-B-C', 'pip', allow_stderr_warning=True
)
```

Reply:
Actually found that the default on black, _version 19.3b0_, is to put it on a single line (as 88 characters are allowed but this would cause a pep8 failure.

Reformatted as suggested and added a check to test_report_single_not_found that it is **only** returning the Warning i.e. no lines on `stdout`.